### PR TITLE
[stable/jenkins] Fix component label in NOTES.txt ...

### LINKED ISSUE #7442: [stable/jenkins] Component label in NOTES.txt does not match name in jenkins-master-deployment.yaml
@lachie83 @viglesiasce 

In NOTES, it uses `{{jenkins.fullname}}` https://github.com/helm/charts/blob/e0a20b0b9bba23a1892c1aa8a41099467861f114/stable/jenkins/templates/NOTES.txt#L21

however it will be labelled as `component: "{{ .Release.Name }}-{{ .Values.Master.Name }}"`  in https://github.com/helm/charts/blob/e0a20b0b9bba23a1892c1aa8a41099467861f114/stable/jenkins/templates/jenkins-master-deployment.yaml#L9

For example, when using `jenkins` as the Release Name:

```
helm install -n jenkins stable/jenkins -f helm/jenkins/values.yaml 
```

The master is created and labelled:

```
Labels:         app=jenkins
                chart=jenkins-0.16.23
                component=jenkins-jenkins-master
                heritage=Tiller
                pod-template-hash=4160751010
                release=jenkins
```

However, NOTES shows:

```
2. Get the Jenkins URL to visit by running these commands in the same shell:
  export POD_NAME=$(kubectl get pods --namespace default -l "component=jenkins-master" -o jsonpath="{.items[0].metadata.name}")
  echo http://127.0.0.1:8080
  kubectl port-forward $POD_NAME 8080:8080
```

And that won't work because the label does not match.

### PR DESCRIPTION
**Pull request by francois-travais**
to match label in jenkins-master-deployment.yaml (#7442)

Signed-off-by: François Travais <francois.travais@gmail.com>

#### What this PR does / why we need it:

Fix component label in NOTES.txt (when service type is ClusterIP) to match label in jenkins-master-deployment.yaml.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #7442 

#### Checklist
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped

### CODE REVIEW

File: stable/jenkins/templates/NOTES.txt [hunk 1]
```diff
@@ -18,7 +18,7 @@
   echo http://$SERVICE_IP:{{ .Values.Master.ServicePort }}/login
 
 {{- else if contains "ClusterIP"  .Values.Master.ServiceType }}
-  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "component={{ template "jenkins.fullname" . }}-master" -o jsonpath="{.items[0].metadata.name}")
+  export POD_NAME=$(kubectl get pods --namespace {{ .Release.Namespace }} -l "component={{ .Release.Name }}-{{ .Values.Master.Name }}" -o jsonpath="{.items[0].metadata.name}")
```
Comment (davidkarlsen) [inline stable/jenkins/templates/NOTES.txt]: actually the pod seems to have label:
"{{ .Release.Name }}-{{ .Values.Master.Component }}"
(There is a difference between the deployment's label and it's containers)
https://github.com/helm/charts/blob/master/stable/jenkins/templates/jenkins-master-deployment.yaml

Comment (francois-travais) [reply stable/jenkins/templates/NOTES.txt]: True, I missed it.

**Pull request merged**